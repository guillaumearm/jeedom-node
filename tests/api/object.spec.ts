import { evolve, identity } from 'ramda'
import { prepareGenericData, prepareJeedomTest } from '../utils'

describe('object', () => {
  const [api, recordTest] = prepareJeedomTest()

  const prepareData = prepareGenericData(
    evolve({
      eqLogics: prepareGenericData(
        evolve({
          cmds: prepareGenericData(identity),
        }),
      ),
    }),
  )

  recordTest('object::all', async () => {
    const objects = await api.object.all()
    expect(prepareGenericData(identity)(objects)).toMatchSnapshot()
  })

  recordTest('object::full', async () => {
    const fullObjects = await api.object.full()
    expect(prepareData(fullObjects)).toMatchSnapshot()
  })

  recordTest('object::fullById', async () => {
    const fullObjects = [await api.object.fullById({ id: 1 })]
    expect(prepareData(fullObjects)).toMatchSnapshot()
  })

  recordTest('[error] object::fullById', () => {
    const response = api.object.fullById({ id: -1 })
    const expectedError = 'Objet introuvable : -1'
    return expect(response).rejects.toThrow(expectedError)
  })
})
