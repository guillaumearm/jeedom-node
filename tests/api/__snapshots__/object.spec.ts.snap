// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object object::all 1`] = `
Array [
  Object {
    "configuration": Object {},
    "display": Object {},
    "father_id": null,
    "id": "13",
    "isVisible": "1",
    "name": "Automatismes",
    "position": null,
  },
]
`;

exports[`object object::full 1`] = `
Array [
  Object {
    "configuration": Object {},
    "display": Object {},
    "eqLogics": Array [
      Object {
        "category": Object {
          "automatism": "1",
          "default": "0",
          "energy": "0",
          "heating": "0",
          "light": "0",
          "multimedia": "0",
          "security": "0",
        },
        "cmds": Array [
          Object {
            "alert": null,
            "configuration": Object {},
            "display": Object {},
            "eqLogic_id": "119",
            "eqType": "jeedouino",
            "generic_type": "GENERIC_ACTION",
            "html": null,
            "id": "1479",
            "isHistorized": "0",
            "isVisible": "1",
            "logicalId": "ID15a",
            "name": "15_Send2LCD",
            "order": "15",
            "subType": "message",
            "template": null,
            "type": "action",
            "unite": "",
            "value": null,
          },
        ],
        "comment": null,
        "configuration": Object {},
        "display": Object {},
        "eqReal_id": null,
        "eqType_name": "jeedouino",
        "generic_type": null,
        "id": "119",
        "isEnable": "1",
        "isVisible": "1",
        "logicalId": "",
        "name": "Serre",
        "object_id": "13",
        "order": "0",
        "status": Object {
          "danger": 0,
          "lastCommunication": "2018-09-20 08:10:08",
          "timeout": 0,
          "warning": 0,
        },
        "timeout": null,
      },
    ],
    "father_id": null,
    "id": "13",
    "isVisible": "1",
    "name": "Automatismes",
    "position": null,
  },
]
`;

exports[`object object::fullById 1`] = `
Array [
  Object {
    "configuration": Object {},
    "display": Object {},
    "eqLogics": Array [
      Object {
        "category": Object {
          "automatism": "0",
          "default": "0",
          "energy": "0",
          "heating": "0",
          "light": "0",
          "multimedia": "0",
          "security": "0",
        },
        "cmds": Array [
          Object {
            "alert": null,
            "configuration": Object {},
            "currentValue": 1,
            "display": Object {},
            "eqLogic_id": "204",
            "eqType": "virtual",
            "generic_type": null,
            "html": null,
            "id": "1782",
            "isHistorized": "0",
            "isVisible": "1",
            "logicalId": "light",
            "name": "Lumière",
            "order": "0",
            "subType": "numeric",
            "template": null,
            "type": "info",
            "unite": "",
            "value": null,
          },
        ],
        "comment": null,
        "configuration": Object {},
        "display": Object {},
        "eqReal_id": null,
        "eqType_name": "virtual",
        "generic_type": null,
        "id": "204",
        "isEnable": "1",
        "isVisible": "0",
        "logicalId": "summary1",
        "name": "Résumé",
        "object_id": "1",
        "order": "0",
        "status": Object {
          "danger": 0,
          "lastCommunication": "2018-11-04 14:43:51",
          "timeout": 0,
          "warning": 0,
        },
        "timeout": null,
      },
    ],
    "father_id": null,
    "id": "1",
    "isVisible": "1",
    "name": "Maison",
    "position": null,
  },
]
`;
